### ISSUE TITLE
`needless_return` and `implicit_return` are opposites

### ISSUE THREAD

**Issue by shenek** [rust-lang/rust-clippy#4197]
```
cargo clippy -V
clippy 0.0.212 (265318d 2019-05-17)
```

Hello I don't think it is possible for a code to pass both `needless_return` and `implicit_return`, because they are opposites to each other.

https://rust-lang.github.io/rust-clippy/master/index.html#needless_return
https://rust-lang.github.io/rust-clippy/master/index.html#implicit_return

So my code won't pass unless i silence one of them.

Moreover example code from `implicit_return` won't compile because it is missing a return type.
```rust
fn foo(x: usize) {
    x
}
```
is supposed to be
```rust
fn foo(x: usize) -> usize {
    x
}
```

**Comment by flip1995** [comment]
`implicit_return` is a restriction lint, meaning that it restricts the language. Some of the restriction lints also trigger on idiomatic rust code, like the `implicit_return` lint. These lints should be enabled only on their own and not the whole group and only if you really want what the lint tells you. For example, during development you probably don't want to enable the `unimplemented` lint. In production, this lint is really useful though.

The documentation should be fixed though. Thanks for catching that! :)
